jgrapht-core/src/main/java/org/jgrapht/traverse/LexBreadthFirstIterator.java

```diff
@@ -559,60 +681,41 @@ void insertBefore(Bucket bucket)
-             *
-             * @return vertex, that was removed from this bucket, null if the bucket was empty.
-             */
-            V poll()
+            V poll(boolean useStarTiebreaking)
             {
                 if (vertices.isEmpty()) {
                     return null;
                 } else {
-                    return vertices.poll();
-                }
-            }
-            
-            /**
-             * Retrieves one vertex from this bucket (according to LBFS*).
-             *
-             * @return vertex, that was removed from this bucket, null if the bucket was empty.
-             */
-            V poll(HashMap<V, Integer> neighborIndexA, HashMap<V, Integer> neighborIndexB, HashMap<V, Set<V>> ASets, HashMap<V, Set<V>> BSets, HashMap<V, Integer> priorityA, HashMap<V, Integer> priorityB) {
-                if (vertices.isEmpty()) {
-                    return null;
-                } else {
-                    V alpha = vertices.peek();
-                    V beta = verticesB.peek();
-                    
-                    if(neighborIndexA.get(alpha) > priorityA.get(alpha)) {
-                        vertices.remove(beta);
-                        return verticesB.poll(); // return Beta
-                    } else if(neighborIndexB.get(beta) > priorityB.get(beta)) {
-                        verticesB.remove(beta);
-                        return vertices.poll(); // return Alpha
-                    } else if(BSets.get(beta).isEmpty() || !ASets.get(alpha).isEmpty()) {
-                        vertices.remove(beta);
-                        return verticesB.poll(); // return Beta
-                    } else if(Objects.equals(neighborIndexA.get(BSets.get(beta).iterator().next()), priorityA.get(alpha))) {
-                        vertices.remove(beta);
-                        return verticesB.poll(); // return Beta
+                    if(!useStarTiebreaking) {
+                        return vertexNumbering.get(vertices.poll());
                     } else {
-                        verticesB.remove(beta);
-                        return vertices.poll(); // return Alpha
+                        int idxAlpha = vertices.peek();
+                        int idxBeta = verticesB.peek();
+                        
+                        V alpha = vertexNumbering.get(idxAlpha);
+                        V beta = vertexNumbering.get(idxBeta);
+                        
+                        if(neighborIndexA.get(alpha) > priorityA.get(alpha)) {
+                            vertices.remove(idxBeta);
+                            return vertexNumbering.get(verticesB.poll()); // return Beta
+                        } else if(neighborIndexB.get(beta) > priorityB.get(beta)) {
+                            verticesB.remove(idxAlpha);
+                            return vertexNumbering.get(vertices.poll()); // return Alpha
+                        } else if(BSets.get(beta).isEmpty() || !ASets.get(alpha).isEmpty()) {
```

Comment:
Saw the BSets and ASets multiple times, but: Objectnames start with lower case :/

Reply:
done